subsys/usb/device_next/class/usbd_cdc_ncm_private.h

```diff
@@ -0,0 +1,127 @@
+/*
```

Comment:
Not sure about this one: if you have a look at https://github.com/zephyrproject-rtos/zephyr/tree/main/subsys/usb/device_next/class you will notice another "private" header which was the reason for me to put the NCM internals there.  Perhaps I should rename it to "usbd_cdc_ncm_internal.h"?

Reply:
Yes although every USB class also seems to have a header in [include](https://github.com/zephyrproject-rtos/zephyr/tree/95dde52b1e85b64bcb1d9ebb8c92dddb5bdc93e1/include/zephyr/usb/class)... Except CDC ECM!

Reply:
But those header contain some kind of API.  The NCM structs are internal and of no interest to externals.  So they should be private/internal.

Reply:
Ah understood! It looks like there is only a special case for CDC only, so maybe it is good to keep the declarations as local as possible...
https://github.com/zephyrproject-rtos/zephyr/blob/95dde52b1e85b64bcb1d9ebb8c92dddb5bdc93e1/include/zephyr/usb/class/usb_cdc.h#L149-L211

I am curious to read the result of it.

Reply:
hmmm... not sure if I got your point...

Most of the descriptor stuff seems to be in usb_ch9.h.  The class headers are one level below of that and contain class dependent stuff of the descriptors plus some API.  So usb_cdc.h with its descriptor definitions is at the right place.

But nevertheless this is not completely clear, because e.g. msos_desc.h is at the same level as usb_ch9.h but clearly belongs to MSC.  So perhaps everything is experimental and needs some cleanup.

Reply:
> But nevertheless this is not completely clear, because e.g. msos_desc.h is at the same level as usb_ch9.h but clearly belongs to MSC. So perhaps everything is experimental and needs some cleanup.

I find it really surprising that you mention that `msos_desc.h` "clearly belong to MSC", because it does not. It has really nothing to do with MSC and MSC device wouldn't benefit at all from `msos_desc.h`. The real benefit for `msos_desc.h` is to allow device to tell Windows that it wants WinUSB driver automatically installed. This is not needed at all for MSC, because MSC uses well-known Mass Storage class driver.

Reply:
Sorry, yes you are right of course.  Mixed several things up and to make it worse I have used in one of my projects this WinUSB mechanism to have "driverless" setup (i.e. without fiddling) under Win10.

Nevertheless the file should be still in the class subdirectory, or not?